File: C++/A/373A-collectingbeats.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int k;
    string panels, panels2, panels3, panels4;
    cin >> k >> panels >> panels2 >> panels3 >> panels4;
    panels += panels2 + panels3 + panels4;
    for (char c : "123456789")
    {
        if (count(panels.begin(), panels.end(), c) > (k * 2))
        {
            cout << "NO";
            return 0;
        }
    }
    cout << "YES";
}

File: Python/373A-collectingbeats.py
k = int(input())
panels = input().strip()
panels2 = input().strip()
panels3 = input().strip()
panels4 = input().strip()
panels += panels2 + panels3 + panels4
flag = True
for i in range(1, 10):
    if (panels.count(str(i)) > (k * 2)):
        print('NO')
        exit()
print('YES')